src/overlays/actors/ovl_En_Ta/z_en_ta.c

```diff
@@ -738,47 +800,57 @@ void func_80B154FC(EnTa* this, PlayState* play) {
 
     if (gSaveContext.timer1Value == 0 && !Play_InCsMode(play)) {
         Audio_QueueSeqCmd(SEQ_PLAYER_BGM_MAIN << 24 | NA_BGM_STOP);
-        this->unk_2E0 &= ~0x200;
+        this->stateFlags &= ~TALON_STATE_FLAG_RESTORE_BGM_ON_DESTROY;
         func_80078884(NA_SE_SY_FOUND);
         gSaveContext.timer1State = 0;
         func_8002DF54(play, &this->actor, 1);
+
+        // Time's up text
         Message_StartTextbox(play, 0x2081, &this->actor);
-        this->actionFunc = func_80B15424;
-        func_80B14E28(this, play);
-        CLEAR_EVENTINF(EVENTINF_HORSES_08);
-        this->unk_2E0 |= 0x80;
+        this->actionFunc = EnTa_TalkCuccoGameEnd;
+        EnTa_CreateFloorCamera(this, play);
+        CLEAR_EVENTINF(EVENTINF_CUCCO_GAME_WON);
+        this->stateFlags |= TALON_STATE_FLAG_FLOOR_CAMERA_ACTIVE;
         Animation_Change(&this->skelAnime, &gTalonSitHandsUpAnim, 1.0f, 8.0f, 29.0f, ANIMMODE_ONCE, -10.0f);
-        this->unk_2E0 &= ~0x10;
-        this->unk_2E0 &= ~0x100;
+        this->stateFlags &= ~TALON_STATE_FLAG_ANIMATION_FINISHED;
+        this->stateFlags &= ~TALON_STATE_FLAG_RAISING_HANDS;
     }
 
-    this->unk_2E0 |= 1;
+    this->stateFlags |= TALON_STATE_FLAG_TRACKING_PLAYER;
 }
 
-void func_80B1585C(EnTa* this, PlayState* play) {
+void EnTa_ThrowSuperCuccos(EnTa* this, PlayState* play) {
     s32 i;
 
-    if (this->unk_2CC > 35) {
+    if (this->timer > 35) {
+        // During the first part of the throw animation,
+        // just turn them (on the table or the floor)
         for (i = 1; i < ARRAY_COUNT(this->superCuccos); i++) {
             if (this->superCuccos[i] != NULL) {
                 Math_SmoothStepToS(&this->superCuccos[i]->actor.world.rot.y, i * -10000 - 3000, 2, 0x800, 0x100);
                 this->superCuccos[i]->actor.shape.rot.y = this->superCuccos[i]->actor.world.rot.y;
             }
         }
-    } else if (this->unk_2CC == 35) {
+    } else if (this->timer == 35) {
+        // At this point, prepare each super cucco for flight
         for (i = 0; i < ARRAY_COUNT(this->superCuccos); i++) {
-            this->unk_2C4[i] = (s32)(Rand_CenteredFloat(6.0f) + 10.0f);
+            // Set a 10-16 frame zero-gravity time for each super cucco
```

Comment:
It's not 10-16 actually, `Rand_CenteredFloat(6)` returns a random number in (-3,3)

with the truncation I think it means the timer can be 7-12 ?